### PR TITLE
Implement InYear and InMonth for date types

### PR DESCRIPTION
**Pull request by Corniel**
As requested by #288, it should be easy to check if a certain date (time) type is in a specific month or year.

I'm not sure yet of it should be `date.In(Year)` and `date.In(Month)` or `date.InYear(Year)` and `date.InYear(Month)`.

### CODE REVIEW

**Comment by ahofland** [comment]
It depends a bit on what the practical application is, but wouldn't you also want to support ints in both functions? There is of course then an assumption (that the year is AC) but normally that's what you want.
Also, for the current implementation, .in(Year) and .in(Month) will be fine, as the type implies the usage. If you'd want to support ints, it'd need to be .InMonth(int) and .InYear(int) otherwise you don't have a way to separate them.

**Comment by Corniel** [comment]
> It depends a bit on what the practical application is, but wouldn't you also want to support ints in both functions? There is of course then an assumption (that the year is AC) but normally that's what you want. Also, for the current implementation, .in(Year) and .in(Month) will be fine, as the type implies the usage. If you'd want to support ints, it'd need to be .InMonth(int) and .InYear(int) otherwise you don't have a way to separate them.

If you take the API of `System.DateTime` you see both `DateTime.Add(TimeSpan)` and `DateTime.AddMonths(int)`, so with that in mind you could argue that `.In(Year)` and `.In(Month)` is the best naming.

I'm not even sure if `.InYear(int)` should be supported: What does it add compared to `DateTime.Year == year`? For the `Year` and `Month` SVO on the otherside, There are downsides on supporting those overloads.

**Comment by pmdevers** [comment]
Mayby we should not add it to the Svo it self but use extention methods. Because it does not add functionality to the svo.

**Comment by Corniel** [comment]
> Mayby we should not add it to the Svo it self but use extention methods. Because it does not add functionality to the svo.

What do you want to gain with that?

File: src/Qowaiv/Extensions/System.DateOnly.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    /// <returns>
+    /// A new date whose value is the sum of the date and time represented
+    /// by this instance and the time interval represented by value.
+    /// </returns>
+    /// <exception cref="ArgumentOutOfRangeException">
+    /// The resulting date is less than <see cref="DateOnly.MinValue"/> or greater
+    /// than <see cref="DateOnly.MaxValue"/>.
+    /// </exception>
+    [Pure]
+    public static DateOnly Add(this DateOnly d, DateSpan value, bool daysFirst)
```
Comment (Sjaaky) [inline src/Qowaiv/Extensions/System.DateOnly.cs:51]: I see how this method solves a problem. But I don't think it makes clear what happens to the reader of calling code.

Comment (Corniel) [reply src/Qowaiv/Extensions/System.DateOnly.cs:51]: This is the same API as for Date (and DateTime). So changing is a breaking change.

File: src/Qowaiv/Date.cs [hunk 2]
```diff
@@ -370,7 +368,19 @@ public Date Add(DateSpan value, bool daysFirst)
     [Pure]
     public Date AddMilliseconds(double value) => new(m_Value.AddMilliseconds(value));
 
-    #endregion
+    /// <summary>Returns true if the date is in the specified month, otherwise false.</summary>
+    /// <param name="month">
+    /// The <see cref="Qowaiv.Month"/> the date should be in.
+    /// </param>
+    [Pure]
+    public bool In(Month month) => !month.IsEmptyOrUnknown() && Month == (int)month;
```
Comment (Sjaaky) [inline src/Qowaiv/Date.cs:376]: I prefer IsIn(...) to emphasize that it is a boolean check.

Comment (Corniel) [reply src/Qowaiv/Date.cs:376]: That is a good suggestion!

**Comment by Corniel** [comment]
> > > Mayby we should not add it to the Svo it self but use extention methods. Because it does not add functionality to the svo.
> > 
> > 
> > What do you want to gain with that?
> 
> Methods that do not change the state of the Svo can be placed outside. I believe this will improve readability of a class.

All SVO's are immutable. All methods are decorated as `[Pure]`, so then all methods would become extension methods. I don't think that would help anything here?!

**Comment by Sjaaky** [comment]
> Methods that do not change the state of the Svo can be placed outside. I believe this will improve readability of a class.

The purpose of extension methods is to easily extend classes that are not under your control, to extend interfaces with common functionality. And could also be used to allow for a nullable 'this'.
Since the Svo is under our control, the best place is in the Svo itself. For readability we can use regions and partial classes.

**Review by Sjaaky — APPROVED** [review]
lgtm

File: src/Qowaiv/Extensions/System.DateOnly.cs (hunk 1 above)
Comment (Corniel) [reply src/Qowaiv/Extensions/System.DateOnly.cs]: This has been addressed with #290 .

**Pull request merged**